packages/worker/src/worker.ts

```diff
@@ -475,12 +476,16 @@ export class Worker {
     // This isn't required for vscode, only for Chrome Dev Tools which doesn't support debugging worker threads.
     // We also rely on this in debug-replayer where we inject a global variable to be read from workflow context.
     if (compiledOptions.debugMode) {
```

Comment:
This condition is not new, but I'm hesitant about it.

1. I dont like that we bind this condition to `debugMode` directly. `debugMode` is like an "higher abstraction level" option, implicitely affecting multiple lower (or internal) options, but it should not be exposed/used after `compileWorkerOptions()`. Shouldn't we avoid the Threaded VM model whenever `workflowThreadPoolSize = 1`? Or should we support some other magic value, let say `workflowThreadPoolSize = 0`, that disable the use of the Threaded VM model?
2. We have almost no tests for `debugMode = true`, and absolutely none coverring the `debugMode && reuseV8Context` case. We should definitely try to cover these sooner than later.

Reply:
We have test-worker-debug-mode that tests this path.
We can't avoid the worker thread, we need it for isolation of the process global hooks.

Reply:
For now, I'd rather not expose these advanced options to users and support debugging in chrome devtools.